CMakeLists.txt

```diff
@@ -108,9 +108,29 @@ add_subdirectory(test)
 add_subdirectory(tools)
 add_subdirectory(apps)
 string(REPLACE " -Wmissing-declarations" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
 if(PYTHON)
+  find_package(Git QUIET)
+  if(GIT_FOUND AND EXISTS "${TACO_PROJECT_DIR}/.git")
+    # Update submodules as needed
+    option(GIT_SUBMODULE "Check submodules during build" ON)
```

Comment:
This is great.  But could we make this submodule checkout stuff globally available?  I'd like to use it for googletest and bats (see #393) too, and I'd like to be able to clone those submodules even when PYTHON is disabled.

Reply:
Sounds good. I will just pull pybind unconditionally then.

Reply:
It looks good to me, thanks!

I don't see a way to mark it as resolved... maybe I am blind?